ClientApp/viking-vault/src/components/UserIcon.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import icon from "./images/user2.png";
+import "./styles.css";
+import {Redirect} from "react-router-dom";
+
+interface IUserIconState {
+  clicked: boolean;
+  redirect: boolean;
+}
+
+class UserIcon extends React.Component<any, IUserIconState> {
+  megaMenu: any;
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      clicked: false,
+      redirect: false
+    };
+    this.megaMenu = React.createRef();
+    this.handleClick = this.handleClick.bind(this);
```

Comment:
You can avoid binding class methods by using arrow functions instead, ``myClassMethod= () => {}``, the ``this`` keyword is binded(not sure if this is even a word lol) differently in those (in this case it refers to the class, which is what you want)